filesmanager/tasks.py

```diff
@@ -0,0 +1,68 @@
+import tempfile
+import zipfile
+
+from celery import shared_task
+from celery.utils.log import get_task_logger
+from django.conf import settings
+from django.core.files.storage import default_storage
+from xmodule.contentstore.django import contentstore
+
+try:
+    from opaque_keys.edx.keys import AssetKey
+except ImportError:
+    AssetKey = None
+
+
+logger = get_task_logger(__name__)
+
+DOWNLOADS_FOLDER = getattr(settings, "FILES_MANAGER_DOWNLOADS_FOLDER", "downloads")
+
+
+@shared_task(bind=True)
+def create_zip_file_task(self, contents):
+    """
+    Create temporary zip file with the contents passed as a parameter
+    and uploads it to the default storage in the downloads folder.
+
+    Returns the url of the zip file in the storage.
+    """
+    task_id = self.request.id
+    fp = tempfile.NamedTemporaryFile(delete=False, suffix=".zip")
+    path = f"{DOWNLOADS_FOLDER}/{task_id}.zip"
+    with zipfile.ZipFile(fp, "w") as ziph:
+        folder_structure = get_folder_structure_from_content(contents)
+        logger.info(folder_structure)
```

Comment:
are we using this?

Reply:
The folder structure is to transform the directories tree to a list to facilitate the ZipFile creation